realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -248,4 +261,49 @@ static synchronized void invokeWithGlobalRefCount(RealmConfiguration configurati
         }
         callback.onResult(totalRefCount);
     }
+
+    /**
+     * Copies Realm database file from Android asset directory to the directory given in the {@link RealmConfiguration}.
+     * Copy is performed only at the first time when there is no Realm database file.
+     *
+     * @param configuration configuration object for Realm instance.
+     * @throws IOException if copying the file fails.
+     */
+    private static void copyAssetFile(RealmConfiguration configuration) {
+        File realmFile = new File(configuration.getRealmFolder(), configuration.getRealmFileName());
+        if (!realmFile.exists()) {
+            InputStream inputStream = null;
+            FileOutputStream outputStream = null;
+            try {
+                inputStream = configuration.getAssetFile();
+                if (inputStream != null) {
+                    outputStream = new FileOutputStream(realmFile);
+                    byte[] buf = new byte[1024];
```

Comment:
1024 might not be the most optimized length, i doubt 4096 would be faster and bigger buffer should be acceptable even on very low-end devices.